src/main/java/uk/ac/ebi/ena/readtools/utils/CramDumpVersionResultsValidator.java

```diff
@@ -0,0 +1,257 @@
+package uk.ac.ebi.ena.readtools.utils;
+
+import htsjdk.samtools.fastq.FastqReader;
+import htsjdk.samtools.fastq.FastqRecord;
+
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.function.Consumer;
+
+public class CramDumpVersionResultsValidator {

```

Comment:
A more descriptive name for the class might be Sam2FastqStatsComparator (I am assuming here that we rename Cram2Fastq as Sam2Fastq). We are comparing stats between two sets of Fastqs generated by (equivalent of) Sam2Fastq.

This class might actually benefit from JCommander with description of the purpose of the class and its input parameters. However, this is definitely a low priority.